jerry-core/ecma/builtin-objects/ecma-builtin-helpers-json.c

```diff
@@ -221,6 +222,72 @@ ecma_builtin_helper_json_create_non_formatted_json (lit_utf8_byte_t left_bracket
   return ecma_make_string_value (result_str_p);
 } /* ecma_builtin_helper_json_create_non_formatted_json */
 
+/**
+ * Function to Set a string token from the given arguments, fill its fields and advances the string pointer
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_json_fill_token_fields_from_string (const lit_utf8_byte_t * str_start_p, lit_utf8_size_t string_size)
+{
+  ecma_json_token_t token;
+  token.current_p = str_start_p;
+  token.end_p = str_start_p + string_size;
+
+  ecma_value_t final_result = ecma_builtin_json_parse_value (&token);
+
+  if (!ecma_is_value_undefined (final_result))
+  {
+    ecma_builtin_json_parse_next_token (&token, false);
+
+    if (token.type != end_token)
+    {
+      ecma_free_value (final_result);
+      final_result = ECMA_VALUE_UNDEFINED;
+    }
+  }
+  return final_result;
+} /*ecma_builtin_json_fill_token_fields_from_string*/
+
+/**
+ * Function to create a json formated string from an object
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_json_string_from_object (const ecma_value_t arg1)
+{
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+  ecma_json_stringify_context_t context;
+  context.occurence_stack_last_p = NULL;
+  context.indent_str_p = ecma_get_magic_string (LIT_MAGIC_STRING__EMPTY);
+  context.property_list_p = ecma_new_values_collection ();
+  context.replacer_function_p = NULL;
```

Comment:
Are you sure that this is good to set (and possible overwrite) the context values here?

Reply:
It is better now.